app/controllers/animals_controller.rb

```diff
@@ -12,35 +12,36 @@ def new
   def create
     @animal = Animal.new(animal_params)
 
-    respond_to do |format|
-      if @animal.save
-        format.html { redirect_to @animal, notice: 'Animal was successfully created.' }
-        format.json { render :show, status: :created, location: @animal }
-      else
-        format.html { render :new }
-        format.json { render json: @animal.errors, status: :unprocessable_entity }
-      end
+    params[:animal][:shl_numbers_codes].split(",").each do |code|
+      @animal.shl_numbers.find_or_initialize_by(code: code)
+    end
```

Comment:
thoughts on extracting this param transformation logic into a private method on this controller?